### PR TITLE
[DEV-9390] Add requests for Signup page

### PR DESCRIPTION
**Pull request by aspirisen**

### CODE REVIEW

File: src/endpoints/Signup/types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export type EmailAvailability = 'available' | 'taken';
```
Comment (oleg-semyonov) [inline src/endpoints/Signup/types.ts:1]: I think this should be an enum 🤔

File: src/endpoints/Billing/types.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export interface SignupRequest {
+    email: string;
+    company_name: string;
+    first_name: string;
+    last_name: string;
+    plan: 'core';
+    currency: 'eur';
+    quantity: number;
+    billing_cycle: 'month';
```
Comment (oleg-semyonov) [inline src/endpoints/Billing/types.ts:9]: `plan`, `currency` and `billing_cycle` shouldn't be hardcoded. We were asked to provide ways to allow to sign up for any plan.

**Review by oleg-semyonov — CHANGES_REQUESTED** [review]

File: src/endpoints/Billing/types.ts (hunk 2 above)
Comment (oleg-semyonov) [reply src/endpoints/Billing/types.ts]: @aspirisen this one still stands. [Here is the code](https://github.com/prezly/prezly/blob/86312d46d76157f2b5d6a9714a13423522af1f83/lib/Prezly/Support/Plans/Plan.php#L28) that describes all supported values for `plan`, `currency` and `billing_cycle`

File: src/endpoints/Billing/types.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+export enum SignupCurrencies {
+    EUR = 'eur',
+    USD = 'usd',
+}
+
+export enum SignupBillingCycles {
+    YEAR = 'year',
+    MONTH = 'month',
+}
```
Comment (marko-ogg) [inline src/endpoints/Billing/types.ts:15]: These are enum values, I wouldn't use plural, so: `SignupCurrency`, `SignupBillingCycle`. Maybe even without the `Signup` prefix.

**Review by marko-ogg — APPROVED** [review]

File: src/endpoints/Billing/types.ts (hunk 3 above)
Comment (aspirisen) [reply src/endpoints/Billing/types.ts:15]: We already have such enums like here https://github.com/prezly/javascript-sdk/blob/master/src/types/Plan.ts
but looks like they do not much with this code https://github.com/prezly/prezly/blob/86312d46d76157f2b5d6a9714a13423522af1f83/lib/Prezly/Support/Plans/Plan.php#L28

can i use enums from shared types, or i have to create my own for signup?

Comment (aspirisen) [reply src/endpoints/Billing/types.ts]: hm, hence i have to create my own enums, thanks)

**Comment by aspirisen** [comment]
@oleg-semyonov could you take a look

File: src/endpoints/Billing/types.ts [hunk 4]
```diff
@@ -1,12 +1,28 @@
...
     company_name: string;
     first_name: string;
     last_name: string;
     quantity: number;
-    plan?: 'core';
-    currency?: 'eur';
-    billing_cycle?: 'month';
+    plan?: SignupPlan;
+    currency?: SignupCurrency;
+    billing_cycle?: SignupBillingCycle;
```
Comment (oleg-semyonov) [inline src/endpoints/Billing/types.ts:25]: Just to be sure - these are not required in the API, right? I saw them being required in the first commits 🤔

**Review by oleg-semyonov — APPROVED** [review]

File: src/endpoints/Billing/types.ts (hunk 4 above)
Comment (aspirisen) [reply src/endpoints/Billing/types.ts:25]: well, technically they are mandatory
image

i thought that should be changed in future?

**Pull request merged**